src/Auth/types.ts

```diff
@@ -0,0 +1,13 @@
+export type User = {
+    readonly id: number;
+    readonly email: string;
+    readonly name: string;
+    readonly lastname: string;
+};
+
+export type AuthService = {
+    readonly register: (email: string, password: string) => Promise<User> | unknown;
+    readonly login: (email: string, password: string) => Promise<User> | unknown;
+    readonly logout: () => void;
+    readonly getCurrentUser: () => User | null;
```

Comment:
Previously you used `unknown`, but `null` is used here, it might be confusing.
Also it's possible to use [`Null Object`](https://refactoring.guru/introduce-null-object) here, but we can name it `AnonymousUser`.

Reply:
I remove `unknown` above and maybe leave `null` for cases when the user is not authorized.

Reply:
Actually,if you want make it explicit, you can define it by union type.

```ts
type AuthenticatedState = {
    readonly isAuthenticated: false;
} |
{
    readonly isAuthenticated: true;
    readonly  user: User;
}

export type AuthService = {
    //...
    readonly getAuthenticatedState: () => AuthenticatedState;
    //...
}
```

Reply:
Also i guess this service will be decomposed in future :)